pages/esmodule.tsx

```diff
@@ -106,7 +83,10 @@ const ESModule = () => {
             typeof totalStaked !== 'undefined'
               ? canFire
                 ? 100
-                : totalStaked.mul(100).div(thresholdAmount).toFixed()
+                : totalStaked
+                    .mul(100)
+                    .div(thresholdAmount || 1)
```

Comment:
Are you sure adding these updates around `thresholdAmount` don't change any of the intended logic?

Reply:
No, i'm not, but i guessed that if thresholdAmount is not defined and we divide by 0 it will give an error.. But probably we need to check

Reply:
I see. Maybe it's better to not render the progress ring at all if `thresholdAmount` is undefined. Progress ring would be kind of useless if we don't know the total progress to be made.

Reply:
makes sense

Reply:
At the end i checked if there is no thresholdAmount, show 0